### PR TITLE
Support framework frames for Marko

### PR DESCRIPTION
**Pull request by jainsneha23**
Associated Issue: #2980 

### Summary of Changes

* Support framework frames for Marko

### Test Plan

Tell us a little a bit about how you tested your patch.

Example test plan:

- [x] Run the debugger locally
- [x] Debug Marko Sample [https://github.com/marko-js-samples/marko-webpack](https://github.com/marko-js-samples/marko-webpack)

### Screenshots/Videos
screen shot 2017-10-19 at 12 06 18 am
screen shot 2017-10-19 at 12 06 58 am

### CODE REVIEW

File: src/utils/frame.js [hunk 1]
```diff
@@ -14,163 +14,113 @@ function getFrameUrl(frame) {
 }
 
 function isBackbone(frame) {
-  return getFrameUrl(frame).match(/backbone/i);
+  return getFrameUrl(frame).match(/backbone/i) && "Backbone";
```
Comment (jasonLaster) [inline src/utils/frame.js]: do we need these: ` && "Backbone";`?

File: src/utils/frame.js [hunk 2]
```diff
@@ -14,163 +14,113 @@ function getFrameUrl(frame) {
...
+    isLodash(frame) ||
+    isEmber(frame) ||
+    isVueJS(frame) ||
+    isRxJs(frame) ||
+    isAngular(frame) ||
+    isRedux(frame) ||
+    isDojo(frame) ||
+    isImmutable(frame) ||
+    isMarko(frame)
+  );
```
Comment (jasonLaster) [inline src/utils/frame.js]: thanks... but i'm not sure if we want to clean up the code yet. I'm not sure what API i want yet :)

Comment (jainsneha23) [reply src/utils/frame.js]: The function had 20 if conditions, giving me linting error of function complexity 20. So, I was forced to do the cleanup. I can revert back. But then I have to disable this particular linting error. Do you have any other suggestion?

Comment (wldcordeiro) [reply src/utils/frame.js]: I think a solution that would be nicer might look something like this. I wouldn't say it needs to be done here in this PR though.  

```js
const libraryMap = {
  react: {
    label: "React",
    pattern: /react/i,
  },
  ...
};

export function getLibraryFromUrl(frame: Frame) {
  const frameUrl = getFrameUrl(frame);
  return Object
    .values(libraryMap)
    .find(({ pattern }) => frameUrl.match(pattern))
    .label; 
}
```

Comment (jasonLaster) [reply src/utils/frame.js]: haha - okay :)

i think @wldcordeiro has a nice refactor... lets do it 😛

Comment (jainsneha23) [reply src/utils/frame.js]: Great. Even I think a map is better. I will refactor this.

Comment (jainsneha23) [reply src/utils/frame.js]: @jasonLaster Refactoring done.

**Pull request merged**